### PR TITLE
Add UI for {imap,smtp}_certificate_checks options

### PR DESCRIPTION
**Pull request by link2xt**
This PR is to solve #1036 issue

### CODE REVIEW

**Comment by link2xt** [comment]
Waiting for #1082 to be merged first.

Also, `configure` function in `deltachat-node/lib/deltachat.js` needs to be extended to handle these options.

**Comment by link2xt** [comment]
Blocked by https://github.com/deltachat/deltachat-node/pull/388 now

File: src/renderer/components/Login.js [hunk 1]
```diff
@@ -206,6 +211,17 @@ export default class Login extends React.Component {
               <option value='starttls'>STARTTLS</option>
               <option value='plain'>{tx('off')}</option>
             </DeltaSelect>
+            <DeltaSelect
+              id='imap_certificate_checks'
+              label={tx('login_imap_certificate_checks')}
```
Comment (cyBerta) [inline src/renderer/components/Login.js]: I think the entry for login_imap_certificate_checks is missing in the sources, or?

**Comment by Jikstra** [comment]
@link2xt can you update the package.json to use the current deltachat-node git master? This way we can test master a couple of days without having a new -node release.

**Comment by link2xt** [comment]
@Jikstra Done. Doing `npm install` and testing now, will try to rebase on top of master then.

File: src/renderer/components/Login.js (hunk 1 above)
Comment (Jikstra) [reply src/renderer/components/Login.js]: Yup seems like the translations (at least experimental/untranslated) are missing.

**Comment by r10s** [comment]
i think there should be only _one_ option for the certificate checks, there is no good reason to be strict on the one but not on the other; i think this was the consens on irc together with @hpk42 @adbenitez and others.

on android, @cyBerta  just did a thing at https://github.com/deltachat/deltachat-android/pull/1096

Screen Shot 2019-11-01 at 12 09 35

technically, the option sets both core options, the current state is driven by `imap_certificate_checks`

**Comment by link2xt** [comment]
Ok, I'll change the interface, add translations and mark it as ready for review once it works.

**Comment by link2xt** [comment]
I have merged two options into one similarly to Android and added strings to `_untranslated_en.json`. By the way translated strings are already on Transifex.

**Comment by Simon-Laux** [comment]
> By the way translated strings are already on Transifex.

looks like we could do a transifex pull again.

File: src/renderer/components/Login.js [hunk 2]
```diff
@@ -57,21 +58,35 @@ export default class Login extends React.Component {
...
     const { mode } = this.props
     const { id, value } = event.target
-    const updatedState = update(this.state, {
-      credentials: {
+    let updatedCredentials;
+    if (id == 'certificate_checks') {
+      // Change to certificate_checks updates certificate checks configuration
+      // for all protocols.
+      updatedCredentials = {
+        ['imap_certificate_checks']: { $set: value },
```
Comment (Jikstra) [inline src/renderer/components/Login.js]: The [] aren't needed in this case i think.

**Pull request merged**